### PR TITLE
feat: Create a new metric reusable component

### PR DESCRIPTION
**Pull request by RulaKhaled**
# Description
New metric component will reflect tests analytics statistics in the dashboard, an ask from design team.

# Notable Changes
- Added 2 more variants to the badge ui component 
- Created new component for the MetricCard

# Screenshots
https://github.com/user-attachments/assets/57f25e66-250a-4b08-b866-f9b816321ace

Closes: [[Gazebo] Create Metric reuseable component](https://github.com/codecov/engineering-team/issues/2414)

# Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. In 2022 this entity acquired Codecov and as result Sentry is going to need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: src/ui/Badge/Badge.tsx [hunk 1]
```diff
@@ -2,6 +2,8 @@ import cs from 'classnames'
 
 const BadgeVariants = {
   default: 'bg-ds-pink-default text-white',
+  danger: 'bg-white text-ds-primary-red border border-ds-primary-red',
```
Comment (ajay-sentry) [inline src/ui/Badge/Badge.tsx:5]: we may need to update these bg colors for dark mode equivalents

Could potentially create a new class in globals.css called "card-background" or similar

Comment (RulaKhaled) [reply src/ui/Badge/Badge.tsx:5]: so just the `bg-white`?

Comment (RulaKhaled) [reply src/ui/Badge/Badge.tsx:5]: i actually don't need it, it's by default white and it inherits the dark mode styles.

**Review by ajay-sentry — APPROVED** [review]
looks good! Thanks for making the dark mode changes

File: src/ui/Badge/Badge.tsx (hunk 1 above)
Comment (ajay-sentry) [reply src/ui/Badge/Badge.tsx]: its always great when we can delete code AND get stuff for free 😂

**Pull request merged**